types/types.go

```diff
@@ -160,3 +161,47 @@ func (m ComMatrix) Contains(cd ComDetails) bool {
 
 	return false
 }
+
+func ToNFTables(m ComMatrix) ([]byte, error) {
+	var tcpPorts []string
+	var udpPorts []string
+	for _, line := range m.Matrix {
+		if line.Protocol == "TCP" {
+			tcpPorts = append(tcpPorts, fmt.Sprint(line.Port))
+		} else if line.Protocol == "UDP" {
+			udpPorts = append(udpPorts, fmt.Sprint(line.Port))
+		}
+	}
+
+	tcpPortsStr := strings.Join(tcpPorts, ", ")
+	udpPortsStr := strings.Join(udpPorts, ", ")
+
+	result := fmt.Sprintf(`#!/usr/sbin/nft -f
+
+	table ip filter {
+		chain OPENSHIFT {
+			# Allow loopback traffic
+			iif lo accept
+	
+			# Allow established and related traffic
+			ct state established,related accept
+	
+			# Allow ICMP
+			ip protocol icmp accept
+
+			# Allow specific TCP and UDP ports
+			tcp dport  { %s } accept
+			udp dport { %s } accept
+
+			# Logging and default drop
+			log prefix "firewall " drop
+		}
+	
+		chain INPUT {
+			type filter hook input priority 0; policy accept;
```

Comment:
OpenShift/OVN-Kubernetes will also create a filter type hooked on input with default priority 0/zero.
Having that chain and this one hooked at input and with the same priority is NOT recommended.

From nftables documentation:

> It's possible to give two base chains the same priority, but there is no guaranteed evaluation order of base chains with identical priority that are attached to the same hook location

https://wiki.nftables.org/wiki-nftables/index.php/Configuring_chains